go/ethadapter/blockprovider.go

```diff
@@ -58,9 +58,15 @@ func (e *EthBlockProvider) StartStreamingFromHeight(height *big.Int) (*host.Bloc
 	return &host.BlockStream{Stream: streamCh, Stop: cancel}, nil
 }
 
-func (e *EthBlockProvider) IsLive(h gethcommon.Hash) bool {
+func (e *EthBlockProvider) IsLive(b *types.Block) bool {
 	l1Head, err := e.ethClient.FetchHeadBlock()
-	return err == nil && h == l1Head.Hash()
+	if err != nil {
+		e.logger.Warn("unable to fetch head eth block - %w", err)
+		return false
+	}
+	// this log message is helpful for visibility on how far behind the block feeding is
+	e.logger.Info("L1 block provider live-monitoring", "currBlock", b.NumberU64(), "head", l1Head.NumberU64())
+	return b.NumberU64() >= l1Head.NumberU64()-1
```

Comment:
Is the minus one here necessary? Perhaps a comment to explain it, since this would mean we consider a new fork to the parent to be "live" I suppose?

Reply:
Yeah fair, was just to give a bit of wiggle-room on a pretty wooly concept but actually I'll put it back so it just checks that the block == headBlock. (Also added the isLatest flag to the log message so if any weird issues creep up with forks and stuff then we'll see it)